src/as-validator.c

```diff
@@ -1924,6 +1924,49 @@ as_validator_check_branding (AsValidator *validator, xmlNode *node)
 	}
 }
 
+/**
+ * as_validator_check_custom:
+ **/
+static void
+as_validator_check_custom (AsValidator *validator, xmlNode *node)
+{
+	g_autoptr(GHashTable) known_keys = NULL;
+	known_keys = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
+
+	for (xmlNode *iter = node->children; iter != NULL; iter = iter->next) {
+		if (iter->type != XML_ELEMENT_NODE)
+			continue;
+
+		g_autofree gchar *node_content = NULL;
+		const gchar *node_name;
+
+		node_name = (const gchar*) iter->name;
+
+		if (g_strcmp0 (node_name, "value") != 0) {
+			as_validator_add_issue (validator, iter, "custom-invalid-tag", node_name);
+			continue;
+		}
+
+		gchar *key_name = NULL;
+		key_name = as_xml_get_prop_value (iter, "key");
```

Comment:
This is missing a `g_autofree`

Reply:
Using this here, causes a crash when added to the table. Probably because it is out of scope.

Reply:
You need to change the `g_hash_table_add` line to `g_hash_table_add (known_keys, g_steal_pointer (&key_name));` as well for this to work without a crash.